plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/SpringPlugin.java

```diff
@@ -86,14 +70,20 @@ public class SpringPlugin {
     @Init
     ClassLoader appClassLoader;
 
-    public void init() {
+    private Class springChangeHubClass;
+
+    public void init() throws ClassNotFoundException {
         LOGGER.info("Spring plugin initialized");
+        springChangeHubClass = Class.forName("org.hotswap.agent.plugin.spring.SpringChangedHub", true, appClassLoader);
+        ReflectionHelper.set(null, springChangeHubClass, "appClassLoader", appClassLoader);
         this.registerBasePackageFromConfiguration();
         this.initBasePackagePrefixes();
     }
 
-    public void init(String version) {
+    public void init(String version) throws ClassNotFoundException {
         LOGGER.info("Spring plugin initialized - Spring core version '{}'", version);
+        springChangeHubClass = Class.forName("org.hotswap.agent.plugin.spring.SpringChangedHub", true, appClassLoader);
+        ReflectionHelper.set(null, springChangeHubClass, "appClassLoader", appClassLoader);
```

Comment:
It seems to me there's no need to use reflection to call SpringChangeHub's methods. The following code simply works.

```java
SpringChangedHub.addChangedXml(url);
SpringChangedHub.addChangedProperty(url);
SpringChangedHub.addChangedClass(clazz);
```

Reply:
This Class runs at agent classLoader.  SpringChangedHub runs at the classloader of app.